### PR TITLE
Address badge labels copy full address

### PR DESCRIPTION
**Pull request by LeeAlephium**
Resolves https://github.com/alephium/desktop-wallet/issues/342, https://github.com/alephium/desktop-wallet/issues/334, https://github.com/alephium/desktop-wallet/issues/302

### CODE REVIEW

File: src/components/Buttons/ClipboardButton.tsx
Comment (LeeAlephium) [inline src/components/Buttons/ClipboardButton.tsx:125]: `CellClipboard` and `CellChildren` must be defined before these template interpolations, otherwise an error occurs, so they appear before the export.

**Comment by mvaivre** [comment]
image

Could we change the tooltip text to "Copy address"? Otherwise we may think that the label text is going to be copied. Thanks!

**Comment by mvaivre** [comment]
In the Address details view, this is now overflowing:

image

(Not sure it's linked to this PR, but it can be the occasion to fix it anyway :))

**Comment by mvaivre** [comment]
image

We want to see the actual address instead of "address".

**Comment by mvaivre** [comment]
Could you change the "copy icon" based on [this](https://github.com/alephium/desktop-wallet/issues/334#issuecomment-1225776605) comment? It seems to be more commonly used across today's apps. Thanks!

**Comment by mvaivre** [comment]
image

The Addresses' table is broken, because the group isn't shown anymore.

**Comment by LeeAlephium** [comment]
> The Addresses' table is broken, because the group isn't shown anymore.

It was actually the address hash missing :DD

Everything's fixed up!

**Comment by LeeAlephium** [comment]
@mvaivre is there anything left I should address?

**Comment by LeeAlephium** [comment]
Certain this also resolves https://github.com/alephium/desktop-wallet/issues/347

**Comment by LeeAlephium** [comment]
@mvaivre rebase done

**Comment by mvaivre** [comment]
Thanks for the fixes. 

One remaining issue (appart from the table alignment thingy mentioned in #347), the action icons in the address details view become smaller as the window width decreases:

image

**Comment by LeeAlephium** [comment]
@mvaivre ready for more comments

**Comment by mvaivre** [comment]
There are still some things itching my eyes a bit :P

Row text baseline isn't consistent :
image

Also, after taking a step back, I think that showing the main address "star icon" doesn't work that well visually and doesn't bring vital info to the table (we don't need to know which is the current main address in TX history tables..). Let's use a "dot" for all addresses.

**Comment by mvaivre** [comment]
Also, could we align the "from / to" adresses and badge so that the spacing is consistent?

image


image

This would make the data easier to parse. Thanks!

**Comment by mvaivre** [comment]
And last little thing while I'm at it. To respect the design evolution steps and keep consistency throughout versions, let's add a border to tables and remove the box shadow.

image

I'll work on an overall UI update taking inspiration from the mobile UI for 1.5.0. 

Thanks a lot! 🙏

**Comment by LeeAlephium** [comment]
@mvaivre review

**Comment by mvaivre** [comment]
Thanks, very nice.

2 little things :

1) Do you think that we could give a bit more space to addresses & labels if we have some?
image

Let's still set a max-width, but bigger than what it is today to show a bit more info on wide screens.
PS: Those columns will be way more useful with middle ellipsis (https://github.com/alephium/desktop-wallet/issues/350)

2) Could you show ellipsis here as well?
image

**Comment by LeeAlephium** [comment]
> PS: Those columns will be way more useful with middle ellipsis

I'll do that then first and use it here

**Comment by LeeAlephium** [comment]
@mvaivre review

I put the space you requested on Slack in the fix-address-cut branch

**Comment by mvaivre** [comment]
Alrighty! I really like the middle ellipsis.

image

Could you try to get those two spaces (1 and 2) equivalent? If not possible, I'd rather have 1 being smaller, as I want to put more emphasis (ie. free space around) the amount. 

Also, as I said in my previous message :

> Let's still set a max-width, but bigger than what it is today to show a bit more info on wide screens.

Let's limit the addresses/labels width to around 200px :

image

**Comment by LeeAlephium** [comment]
> Could you try to get those two spaces (1 and 2) equivalent? 

Just to be sure, then the other line with "Second" should have very large spaces on the sides (and cut off a large portion of the other label)?

**Comment by LeeAlephium** [comment]
Yeah, looks quite nice to me:

File: src/components/Buttons/ClipboardButton.tsx
Comment (nop33) [inline src/components/Buttons/ClipboardButton.tsx:99]: In which case will `children` be falsy AND `Array.isArray(children) && children.length > 0` be true? In other words, when will the condition be met?

Comment (LeeAlephium) [reply src/components/Buttons/ClipboardButton.tsx:99]: Never since `children.length > 0` excludes `[]` being a possibility

Comment (LeeAlephium) [reply src/components/Buttons/ClipboardButton.tsx:99]: I'm also confused by the question, the code is "is children truthy OR (is children a non-empty list)"...

Comment (nop33) [reply src/components/Buttons/ClipboardButton.tsx]: I changed it. Resolving.

**Review by nop33 — APPROVED** [review]
Code review and functional review is done, all good on my side. I would not merge it until @mvaivre also gives his approval.

Well done! 👏

**Review by mvaivre — APPROVED** [review]

**Pull request merged**
